### PR TITLE
[Enhancement] combined count when loadfile in dpp

### LINKED ISSUE #14292: [SparkLoad] dpp read parquet/orc optimization
## Enhancement

Current SparkDpp will execute `spark.read().orc/parquet (single_file)` and `sourceData.count()` for each parquet file, each file will generate a job, and the jobs are serialized. If there are many files, it will take a long time.

Suggested optimizations:
read parquet files under a file group together, and need to consider the scenario of reading partition column from the path.

### PR DESCRIPTION
**Pull request by MaxWk**
Fixes #14292

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/etl/EtlJobConfig.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 import java.util.List;
 import java.util.Map;
 
-/**
+    /**
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/etl/EtlJobConfig.java:34]: revert this change

**Review by wyb — APPROVED** [review]

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java [hunk 2]
```diff
@@ -976,32 +976,53 @@ private Dataset<Row> loadDataFromFilePaths(SparkSession spark,
...
                     }
                 }
+
             } catch (Exception e) {
                 LOG.warn("parse path failed:" + filePath);
                 throw e;
             }
         }
+
+        if (fileGroup.fileFormat != null &&
```
Comment (Astralidea) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/SparkDpp.java:1020]: nit:
```
Set<String> formats = new HashSet<>(Arrays.asList("orc", "parquet"));
if (fileGroup.fileFormat != null && formats.contains(fileGroup.fileFormat.toLowerCase())) {
    scannedRowsAcc.add(fileGroupDataframe.count());
}
```

**Review by Astralidea — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5
